File: BookServiceJava/src/main/java/org/elcata98/mybooks/booksservice/persistence/UserRepository.java
package org.elcata98.mybooks.booksservice.persistence;

import org.elcata98.mybooks.booksservice.model.User;
import org.springframework.stereotype.Repository;

@Repository
public interface UserRepository extends EntityRepository<User> {

    @Override
    default boolean willAccept(Class<?> clazz) {
        return User.class == clazz;
    }
}

File: BooksService/src/main/kotlin/org/elcata98/mybooks/booksservice/persistence/UserRepository.kt
package org.elcata98.mybooks.booksservice.persistence

import org.elcata98.mybooks.booksservice.model.User
import org.springframework.data.repository.Repository
import java.util.UUID

interface UserRepository : Repository<User, UUID> {

    fun save(user: User): User
    fun findByUserId(userId: UUID): User
    fun delete(user: User)
}